### ISSUE TITLE
Functions are incorrectly flagged as undocumented if not generating HTML or LaTeX

### ISSUE THREAD

**Issue by SRGDamia1** [doxygen/doxygen#11147]
**Describe the bug**
A simple function with complete documentation is flagged as not having a documented parameter or return type if HTML or LaTeX is not generated.

**To Reproduce**

Attached: 
[DoxyOutputTest.zip](https://github.com/user-attachments/files/17066956/DoxyOutputTest.zip)

Create a simple header file with a single documented function:
```cpp
/**
 * @file text.h
 */

/**
 * @brief a test function
 *
 * @param input_param an input param
 * @return The return value
 */
uint8_t aTestFxn(uint8_t input_param);

```

enable all warnings:
```
WARNINGS               = YES
WARN_IF_UNDOCUMENTED   = YES
WARN_IF_DOC_ERROR      = YES
WARN_IF_INCOMPLETE_DOC = YES
WARN_NO_PARAMDOC       = YES
WARN_IF_UNDOC_ENUM_VAL = YES
WARN_LOGFILE           = warnings.txt
WARN_AS_ERROR          = FAIL_ON_WARNINGS_PRINT
```

enable XML output and **NOT** HTML or LaTeX:
```
GENERATE_HTML          = NO
GENERATE_LATEX         = NO
GENERATE_XML           = YES
```

get a warning for undocumented parameters and returns:
```log
...text.h:8: warning: parameters of member aTestFxn are not documented
...text.h:8: warning: return type of member aTestFxn is not documented
```

re-enable either the LaTeX *or* the HTML output
```
GENERATE_HTML          = NO
GENERATE_LATEX         = YES
GENERATE_XML           = YES
```

No warnings are produced.

**Expected behavior**
There should not be errors listed for properly documented functions when only XML is generated.

I also tested a function with no documentation, with partial, and with incorrectly formatted documentation, and those same to give me the expected errors regardless of which output I chose. (See the attached zip.)

**Version**

1.12.0 (c73f5d30f9e8b1df5ba15a1d064ff2067cbb8267) on Windows 11

**Comment by albert-github** [comment]
It looks like the problem was introduced in doxygen 1.9.4, more precise:
2c76059d714ea6f2a179578966021df9e365a2a6 is the first bad commit
```
commit 2c76059d714ea6f2a179578966021df9e365a2a6 (HEAD)

Date:   Wed Apr 20 19:52:43 2022 +0200

    Minor optimisation

 src/docparser_p.h  |  4 ++++
 src/outputlist.cpp | 23 +++++++++++++----------
 2 files changed, 17 insertions(+), 10 deletions(-)
```

I've just pushed a proposed patch, pull request #11148 
(Note there is also an extra comment with the proposed pull request regarding no output format selected at all)

**Comment by SRGDamia1** [comment]
Thank you for looking into this so quickly!

**Comment by albert-github** [comment]
Proposed pull request #11148 is made obsolete by proposed pull request #11151 (see discussion with #11148)

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
